core/src/UI/CoreUIParent.cs

```diff
@@ -42,6 +45,60 @@ public CoreUIParent()
         internal bool UseEmbeddedWebview { get; set; }
 #endif
 
+#if iOS
+        /// <summary>
+        /// Caller UIViewController
+        /// </summary>
+        public UIViewController CallerViewController { get; set; }
+
+        internal UIViewController FindCurrentViewController(UIViewController CallerViewController)
+        {
+            if (CallerViewController is UITabBarController)
+            {
+                UITabBarController tabBarController = (UITabBarController)CallerViewController;
+                return FindCurrentViewController(tabBarController.SelectedViewController);
```

Comment:
>FindCurrentViewController [](start = 23, length = 25)

what will happen here? Will it not just return what you pass as parameter? (same for below)... is recursive needed?

Reply:
I didn't change the implementation from what was alreayd happening. I'd have to read up on the UIViewController to see if this is the right way to do it.